store/src/main/java/com/nytimes/android/external/store/base/AllPersister.java

```diff
@@ -0,0 +1,27 @@
+package com.nytimes.android.external.store.base;
+
+
+import java.io.FileNotFoundException;
+
+import javax.annotation.Nonnull;
+import javax.annotation.OverridingMethodsMustInvokeSuper;
+
+import rx.Observable;
+
+public interface AllPersister<Raw> extends DiskAllRead, DiskAllErase {
```

Comment:
We should also implement `Persister` here, so that users can pass an `AllPersister` to `StoreBuilder.persister()` and access the traditional `read()` and `write()` methods in addition to these new all methods.

Then in `SourceAllPersister`, add an FSReader and FSWriter that will use your new `BarCodeReadAllPathResolver` for those `read(barcode)` and `write(barcode)` operations (similar to how `FileSystemPersister` works).